src/tools/interop/idt/capture/base.py

```diff
@@ -71,20 +84,29 @@ def __init__(
     async def start_capture(self) -> None:
         """
         Start the capture
+        Platform is already started
         """
         raise NotImplementedError
 
     @abstractmethod
     async def stop_capture(self) -> None:
         """
-        Stop the capture
+        Stop the capture and pull any artifacts from remote devices
+        Platform is already stopped
         """
         raise NotImplementedError
 
     @abstractmethod
     async def analyze_capture(self) -> None:
         """
         Parse the capture and create + display helpful analysis artifacts that are unique to the ecosystem
-        Write analysis artifacts to artifact_dir
+        Must be async aware and not interact with stdin
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    async def probe_capture(self) -> None:
+        """
+        Probe the local environment, e.g. ping relevant remote services and write respective artifacts
```

Comment:
Is the list of services platform dependent?

Reply:
They might be, but its up to the ecosystem to handle that as needed.

Reply:
Please re open if needed!